fix!: change all_* and any_* to take *apps instead of a list (#116)
canonical/jubilant: docs/how-to/migrate-from-pytest-operator.md

def test_active(juju: jubilant.Juju):
    juju.deploy('mycharm')
    juju.wait(jubilant.all_active)

>     # Or wait more specifically for only 'mycharm' to be active:

Reviewer: How about explicitly saying that other apps are ignored (saying "only mycharm" could be interpreted as meaning that mycharm is the only active app):

```suggestion
    # Or specifically wait for 'mycharm' to be active (ignoring other applications):
```

Author: Changed to "Or wait for just 'mycharm' to be active (ignore other apps):" due to width/space constraints.

>     # Or wait for just 'mycharm' to be active (ignore other apps):

Reviewer 2: ```suggestion
    # Or wait for just 'mycharm' to be active (ignoring other deployed apps):
```

or "other apps in this model"

Author: This one I don't think is necessary, as it's always "in this model" and apps are always deployed.

It's common to use a `lambda` function to customize the callables further. For example, to wait specifically for `mysql` and `redis` to go active:

```python

> juju.wait(lambda status: jubilant.all_active(status, 'mysql', 'redis'))

Reviewer 2: Would this be a good place to promote composition?

```py
juju.wait(lambda status:
    jubilant.all_active(status, 'mysql', 'redis') and
    jubilant.all_blocked(status, 'foobar'))
```

Author: Good idea -- done, and updated the text slightly to mention that
